src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -43,24 +48,41 @@ class TaskStatusUpdateProcessorImpl @Inject() (
 
     val maybeTask = taskTracker.fetchTask(taskId.getValue)
 
-    processUpdate(
-      timestamp = now,
-      appId = appId,
-      maybeTask = maybeTask,
-      mesosStatus = status
-    )
+    maybeTask match {
+      case Some(task) =>
+        processUpdate(
+          timestamp = now,
+          appId = appId,
+          task = task,
+          mesosStatus = status
+        ).map(_ => acknowledge(status))
+      case None =>
+        killUnknownTaskTimer {
+          killTask(taskId)
+          acknowledge(status)
+          Future.successful(())
+        }
+    }
+  }
+
+  private[this] def acknowledge(taskStatus: TaskStatus): Unit = {
+    driverHolder.driver.foreach(_.acknowledgeStatusUpdate(taskStatus))
+  }
+
+  private[this] def killTask(taskId: TaskID): Unit = {
+    driverHolder.driver.foreach(_.killTask(taskId))
```

Comment:
Not related to this PR: could the DriverHolder implement the Scheduler interface by delegating to the real driver or handling the non existence? I see a lot of noise by the optional driver and the handling of non existence driver...